build/patternfly-cli/pf.js

```diff
@@ -66,13 +66,21 @@ program
       nameCapitalized: inflection.capitalize(name),
       bemName: `${bemName}`,
       bemModifierBlock: `{{#if ${bemEntity}--modifier}} {{${bemEntity}--modifier}}{{/if}}`,
+      bemItemModifierBlock: `{{#if ${bemEntity}-item--modifier}} {{${bemEntity}-item--modifier}}{{/if}}`,
       moduleName,
       partialBlock: '{{> @partial-block}}',
-      attributeBlock: `{{#if ${bemEntity}--attribute}}
+      attributeBlock: `
+  {{#if ${bemEntity}--attribute}}
     {{{${bemEntity}--attribute}}}
   {{/if}}`,
-      moduleHbOpen: `{{#> ${dasherizedName}}}`,
+      attributeItemBlock: `id="{{${dasherizedName}-id}}-{{${dasherizedName}-item--id}}"
```

Comment:
A couple of thoughts about this way of defining an `id` attribute on the generated markup of the `__item` instead of using `concat()`.

If we're going to do it this way, I think we should wrap it in a conditional so it doesn't generate an `id` by default, otherwise I think it would be confusing to consumers and react devs that we have `id`'s on elements that aren't used for anything.

Also it's a bit inconsistent if the `__item` generates an ID this way but the main component hbs partial doesn't. We've run into issues before where we've used `(concat 'id="' component--id '-item"')` on a partial when it supported `component-item--id`, generating 2 `id`'s - since it's generally the case that you specify an `id` using `concat()`. I think it would be good to have a consistent approach instead of having to check hbs components for which way it supports adding an `id` to the element in the partial.

If we want to be consistent with this new approach and update the main component hbs to also support generating an `id` in the hbs from the `component--id` attribute, that will create an unnecessary `id` on the main component element since we'll likely always have `component--id` defined, which I think we should avoid.

Reply:
@mcoker try that out